hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -501,6 +507,115 @@ public void setPathProperties(final Path path,
     }
   }
 
+
+  /**
+   * Wrapper on {@link #getBlobProperty(Path, TracingContext)} with the handling
+   * for the httpStatusCode = 404 on the server response.
+   *
+   * @param blobPath path for which the property information is required
+   * @param tracingContext object of TracingContext required for the tracing of
+   * server calls
+   * @return instance of BlobProperty if the blob is present on the given path.
+   * <code>null</code> if there is no blob on the given path.
+   * @throws AzureBlobFileSystemException exception other than
+   * {@link AbfsRestOperationException} for httpStatusCode = 404 on the server
+   * response.
+   */
+  BlobProperty getBlobPropertyWithNotFoundHandling(Path blobPath,
+      TracingContext tracingContext) throws AzureBlobFileSystemException {
+    try {
+      return getBlobProperty(blobPath, tracingContext);
+    } catch (AbfsRestOperationException ex) {
+      if (ex.getStatusCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+        return null;
+      }
+      throw ex;
+    }
+  }
+
+  /**
+   * Calls {@link AbfsClient#getBlobProperty(Path, TracingContext)} on the given
+   * path. Extract the headers from the server-response and converts it to an object
+   * of {@link BlobProperty}.
+   *
+   * @param blobPath blobPath for which property information is requried
+   * @param tracingContext object of TracingContext required for tracing server calls.
+   * @return BlobProperty for the given path
+   * @throws AzureBlobFileSystemException exception thrown from
+   * {@link AbfsClient#getBlobProperty(Path, TracingContext)} call
+   */
+  private BlobProperty getBlobProperty(Path blobPath,
+      TracingContext tracingContext) throws AzureBlobFileSystemException {
+    AbfsRestOperation op = client.getBlobProperty(blobPath, tracingContext);
+    BlobProperty blobProperty = new BlobProperty();
+    final AbfsHttpOperation opResult = op.getResult();
+    blobProperty.setIsDirectory(opResult
+        .getResponseHeader(X_MS_META_HDI_ISFOLDER) != null);
+    blobProperty.setUrl(op.getUrl().toString());
+    blobProperty.setCopyId(opResult.getResponseHeader(X_MS_COPY_ID));
+    blobProperty.setPath(blobPath);
+    blobProperty.setCopySourceUrl(opResult.getResponseHeader(X_MS_COPY_SOURCE));
+    blobProperty.setStatusDescription(
+        opResult.getResponseHeader(X_MS_COPY_STATUS_DESCRIPTION));
+    blobProperty.setCopyStatus(getCopyStatus(opResult));
+    blobProperty.setContentLength(
+        Long.parseLong(opResult.getResponseHeader(CONTENT_LENGTH)));
+    return blobProperty;
+  }
+
+  @org.apache.hadoop.classification.VisibleForTesting
+  String getCopyStatus(final AbfsHttpOperation opResult) {
```

Comment:
Wanted this for running tests in rename API changes. Wherein, we want to check the copyStatus at the destination in case the copy takes some time.
I can change the name of the method, but would want this functionality. Would you suggest removing it completely?

Reply:
taken. will fix the test in rename pr whereever used.